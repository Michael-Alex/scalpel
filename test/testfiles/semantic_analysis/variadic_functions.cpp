struct s
{
	s(int, ...)
	{
	}

	bool
	f(...);
};

void
f(...)
{
}

void
g(double ...);

