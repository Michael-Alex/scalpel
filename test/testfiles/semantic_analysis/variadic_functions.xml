<namespace id="n0">
 <class name="s" id="c0">
  <constructor access="public" variadic="true">
   <parameters>
    <parameter>
     <type>
      <fundamental_type type="int"/>
     </type>
    </parameter>
   </parameters>
  </constructor>
  <destructor access="public" defined="false">
  </destructor>
  <simple_function name="f" access="public" variadic="true" defined="false">
   <return_type>
    <fundamental_type type="bool"/>
   </return_type>
  </simple_function>
 </class>
 <simple_function name="f" variadic="true">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
 </simple_function>
 <simple_function name="g" variadic="true" defined="false">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <fundamental_type type="double"/>
    </type>
   </parameter>
  </parameters>
 </simple_function>
</namespace>
