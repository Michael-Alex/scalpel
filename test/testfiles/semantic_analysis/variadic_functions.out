{
  "id": "namespace-0",
  "classes":
  [
    {
      "id": "class-0",
      "name": "s",
      "constructors":
      [
        {
          "access": "public",
          "variadic": true,
          "inline": true,
          "defined": true,
          "parameters":
          [
            {
              "type":
              {
                "fundamental type": "int"
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "simple functions":
      [
        {
          "id": "simple-member-function-0",
          "name": "f",
          "access": "public",
          "variadic": true,
          "defined": false,
          "return type":
          {
            "fundamental type": "bool"
          }
        }
      ]
    }
  ],
  "simple functions":
  [
    {
      "id": "simple-function-0",
      "name": "f",
      "variadic": true,
      "return type":
      {
        "fundamental type": "void"
      }
    },
    {
      "id": "simple-function-1",
      "name": "g",
      "variadic": true,
      "defined": false,
      "return type":
      {
        "fundamental type": "void"
      },
      "parameters":
      [
        {
          "type":
          {
            "fundamental type": "double"
          }
        }
      ]
    }
  ]
}
